docs/notary-server.md

```diff
@@ -0,0 +1,188 @@
+
+Notary Signer supports mutual authentication - when you generate client
+certificates for your deployment of Notary Server, please make
+sure that the certificates **are not CAs**.  Otherwise if the server is
+compromised, it can sign any number of other client certs.
+
+As an example, please see [this script](opensslCertGen.sh) to see how to
+generate client SSL certs with basic constraints using OpenSSL.
+
+### How to configure and run Notary Server
+
+A JSON configuration file is used to configure Notary Server.  Please see the
+[Notary Server configuration document](notary-server-config.md)
+for more details about the format of the configuration file.
+
+You can also override the parameters of the configuration by
+setting environment variables of the form `NOTARY_SERVER_var`.
+
+`var` is the ALL-CAPS, `"_"`-delimited path of keys from the top level of the
+configuration JSON.
+
+For instance, if you wanted to override the storage URL of the Notary Server
+configuration:
+
+```json
+"storage": {
+	"backend": "mysql",
+	"db_url": "dockercondemo:dockercondemo@tcp(notary-mysql)/dockercondemo"
+}
+```
+
+the full path of keys is `storage -> db_url`. So the environment variable you'd
+need to set would be `NOTARY_SERVER_STORAGE_DB_URL`.
+
+For example, if running the binary:
+
+```
+$ export NOTARY_SERVER_STORAGE_DB_URL=myuser:mypass@tcp(my-db)/dbname?parseTime=true
+$ NOTARY_SERVER_LOGGING_LEVEL=info notary-server -config /path/to/config.json
+```
+
+Note that you cannot override a key whose value is another map.
+For instance, setting
+`NOTARY_SERVER_STORAGE='{"storage": {"backend": "memory"}}'` will not
+set in-memory storage.  It just fails to parse.  You can only override keys
+whose values are strings or numbers.
+
+#### Running Notary Server
+
+Configuration options:
+
+- `-config=<config file>` - The JSON configuration file.
+
+- `-debug` - Passing this flag enables the debugging server on `localhost:8080`.
+	The debugging server provides [pprof](https://golang.org/pkg/net/http/pprof/)
+	and [expvar](https://golang.org/pkg/expvar/) endpoints.
+
+
+Get the official Docker image, which comes with [some sane defaults](
+https://github.com/docker/notary/blob/master/fixtures/server-config-local.json),
+which include a remote trust service but local in-memory backing store.
+
+You can override the default configuration with environment variables.
+For example, if you wanted to run it with just a local signing service instead
+(not recommended for production):
+
+```
+$ docker pull docker.io/docker/notary-server
+$ docker run -p "4443:4443" \
+	-e NOTARY_SERVER_TRUST_SERVICE_TYPE=local
+	notary-server
+```
+
+Alternately, you can run the image with your own configuration file entirely.
+You just need to mount your configuration directory, and then pass the path to
+that configuration file as an argument to the `docker run` command:
+
+```
+$ docker run -p "4443:4443" -v /path/to/config/dir/on/host:/path/in/container \
```

Comment:
I think it's ok to be prescriptive of a path in container here. /etc/docker/notary-server/config.json maybe?